packages/actions/test/unit/security.test.ts

```diff
@@ -89,70 +110,281 @@ describe("Security", () => {
+                )
+            })
+            /// @note If a token has been invalidated, this shuold not allow to access Firebase again
+            /// @todo might not be able to test this in code since it requires revoking access on GitHub
+            it.skip("should not be able to authenticate with a token after this is invalidated", async () => {
+                const auth = createOAuthDeviceAuth({
+                    clientType,
+                    clientId,
+                    scopes: ["gist"],
+                    onVerification: simulateOnVerification
+                })
+                const { token } = await auth({
+                    type: tokenType
+                })
+                // Get and exchange credentials.
+                const userFirebaseCredentials = GithubAuthProvider.credential(token)
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+                const user = getCurrentFirebaseAuthUser(userApp)
+                userId = user.uid
 
-        it("should throw an error if an authenticated user tries to read another user's data", async () => {
-            // login as user2
-            await signInWithEmailAndPassword(userAuth, user2.data.email, user2Pwd)
-            assert.isRejected(getDocumentById(userFirestore, commonTerms.collections.users.name, user1.uid))
+                await signOut(userAuth)
+
+                // @todo how to revoke the token programmatically?
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+            })
+            /// @note A malicious user should not be able to create multiple malicious accounts
+            /// to spam a ceremony
+            // @todo requires adding the checks to the cloud function
+            it("should prevent a user with a non reputable GitHub account from authenticating to the Firebase", async () => {})
+            /// @note If a coordinator disables an account, this should not be allowed to authenticate
+            it("should prevent a disabled account from loggin in (OAuth2)", async () => {
+                const auth = createOAuthDeviceAuth({
+                    clientType,
+                    clientId,
+                    scopes: ["gist"],
+                    onVerification: simulateOnVerification
+                })
+                const { token } = await auth({
+                    type: tokenType
+                })
+                // Get and exchange credentials.
+                const userFirebaseCredentials = GithubAuthProvider.credential(token)
+                await signInToFirebaseWithCredentials(userApp, userFirebaseCredentials)
+
+                const user = getCurrentFirebaseAuthUser(userApp)
+                userId = user.uid
+                // Disable user.
+                const disabledRecord = await adminAuth.updateUser(user.uid, { disabled: true })
```

Comment:
Is there a test to check non admin users cannot update other user records / otherwise disable through an API?

Reply:
The way to access user records is with an [Admin App](https://github.com/quadratic-funding/mpc-phase2-suite/blob/dev/packages/actions/test/utils/configs.ts#L24) instance. In order to get such object back when authenticating to Firebase one would need the `serviceAccountKey.json` file. The [user app](https://github.com/quadratic-funding/mpc-phase2-suite/blob/dev/packages/actions/test/utils/configs.ts#L52) instance (the one you can generate with the firebase API key, which will be published for contributors to use) does not expose methods to alter user accounts

Reply:
I have in mind to do some research on the [REST API](https://cloud.google.com/firestore/docs/use-rest-api) though, as this might expose some interesting test cases